src/test/scala/mesosphere/marathon/api/v2/json/AppUpdateTest.scala

```diff
@@ -210,53 +170,6 @@ class AppUpdateTest extends UnitTest with ValidationTestLike {
       assert(changed == app.copy(acceptedResourceRoles = Set("b")))
     }
 
-    "AppUpdate with a version and other changes are not allowed" in {
```

Comment:
I think we have to make sure that there are still validations error with such updates. See my comment above.

Reply:
Good catch! I added a test for this. I checked other parts too and it seems they cover everything.

Reply:
I can see

1. `assert(vfe.failure.violations.toString.contains("The 'version' field may only be combined with the 'id' field."))`
2. `assert(vfe.failure.violations.toString.contains("may not be set in conjunction with fetch"))`
3. `assert(vfe.failure.violations.toString.contains("cannot specify both ports and port definitions"))`
4. `assert(vfe.failure.violations.toString.contains("ports must be unique"))`

1 is covered in `AppResourceTest`. 4 is covered in `AppValidationTest`. Where do you test 2 and 4? I think we do have the validations but I'd like to confirm this with tests.